### PR TITLE
CB-18344: Extend nameserver ip validation with null check

### PR DESCRIPTION
**Pull request by tiborpopovics**
In this commit I added validation to prevent the nameserver settings in the service pillar from null values. Sometimes we set the nameserver address in the unbound config to null and this causes cluster creation failures.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/decorator/NameserverPillarDecorator.java [hunk 1]
```diff
@@ -56,19 +58,38 @@ private void decorateServicePillarWithDatalakeNameservers(StackDto stackDto, Map
...
                 .map(InstanceMetadataView::getPrivateIp)
-                .filter(StringUtils::hasText)
-                .collect(Collectors.collectingAndThen(Collectors.toList(), Optional::of))
-                .filter(ipList -> !ipList.isEmpty())
-                .orElseThrow(() -> new CloudbreakServiceException("Unable to setup nameservers because there is no IP address present."));
+                .filter(filterValidIps())
+                .collect(Collectors.toList());
+    }
+
+    private static Predicate<String> filterValidIps() {
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/decorator/NameserverPillarDecorator.java:83]: nit: why is this static?

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/decorator/NameserverPillarDecorator.java [hunk 2]
```diff
@@ -56,19 +58,38 @@ private void decorateServicePillarWithDatalakeNameservers(StackDto stackDto, Map
...
+        return ip -> StringUtils.hasText(ip) && !"null".equals(ip);
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/decorator/NameserverPillarDecorator.java:84]: you could use an IPv4 matcher which would be more safe then just checking for `null`

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/decorator/NameserverPillarDecorator.java [hunk 3]
```diff
@@ -56,19 +58,38 @@ private void decorateServicePillarWithDatalakeNameservers(StackDto stackDto, Map
...
             List<String> ipList = getDatalakeGatewayPrivateIps(gatewayInstanceMetadata);
+            validateIpList(ipList);
             servicePillar.put("forwarder-zones", new SaltPillarProperties("/unbound/forwarders.sls",
                     singletonMap("forwarder-zones", singletonMap(datalakeDomain, singletonMap("nameservers", ipList)))));
         } else {
             LOGGER.debug("Skip to add nameserver config for pillar because Datalake stack is not present.");
         }
     }
 
+    private static List<String> getKerberosNameServerIps(KerberosConfig kerberosConfig) {
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/decorator/NameserverPillarDecorator.java:69]: why is this static?

**Review by lacikaaa — CHANGES_REQUESTED** [review]

**Comment by lacikaaa** [comment]
as 2.63 in prod, could you move this pr too?

**Review by lacikaaa — APPROVED** [review]

**Comment by lnardai** [comment]
@keyki 
Hmm. Without the branch guardians approval?
Sounds like a revert

**Pull request merged**